frontend/.eslintrc.js

```diff
@@ -1,17 +1,41 @@
 module.exports = {
-  root: true,
-  parser: '@typescript-eslint/parser',
-  plugins: [
-    '@typescript-eslint',
-  ],
-  extends: [
-    'eslint:recommended',
-    'plugin:@typescript-eslint/recommended',
-    'plugin:react/recommended',
-  ],
-  settings: {
-    react: {
-      version: 'detect'
-    }
-  }
+    parserOptions: {
+        ecmaVersion: 11,
+    },
+    extends: [
+        "eslint:recommended",
+    ],
+    rules: {
+        "indent": "error",
+        "quotes": ["error", "double", { avoidEscape: true }],
+        "semi": "error",
+        "comma-dangle": ["error", "always-multiline"],
+    },
+    overrides: [{
+        files: ["./*"],
+        env: {
+            node: true,
+        },
+    }, {
+        files: ["src/**/*.ts{,x}"],
+        parser: "@typescript-eslint/parser",
+        parserOptions: {
+            tsconfigRootDir: __dirname,
+            project: "./tsconfig.json",
+        },
+        extends: [
+            "plugin:@typescript-eslint/recommended",
+            "plugin:@typescript-eslint/recommended-requiring-type-checking",
+            "plugin:react/recommended",
+        ],
+        settings: {
+            react: {
+                version: "detect",
+            },
+        },
+    }],
+    ignorePatterns: [
+        "/build",
+        "!.*",
+    ],
 };
```

Comment:
Why 4 spaces indent in this file?

Oh wait, is everything 4 spaces now? I thought JS community would prefer 2?

Reply:
Yes, everything is 4 now, with this.

JS is more like C++ in this regard, I think; "the JS community" is not very unified when it comes to certain style things. I picked 4 here mainly because:

* The Rust code has 4, so maybe consistency would be nice, here.
* `App.tsx` already had 4, as well. In fact, only the tooling related JS files had 2.
  * Sure, that's JS, the other one is TS, but TS is a superset of JS and I think, again because of consistency, we shouldn't use different styles for these.

Reply:
Ah I see. The 4 space in `App.ts` was carelessness on my part, not a deliberate decision. Consistency with Rust would be nice, indeed. But I also fear that JSX constructs might naturally be pretty deeply nested and would suffer from 4 spaces indent. But in the end, I don't care I think. But yeah, JS/TS should use the same indentation in our project.

Reply:
JSX is a good point, although you might also argue that deeply nested JSX is an antipattern and a signal to factor out more components.

Also, writing something like

```js
<div><div><div>
    // more components, but only one level deep :ok_hand: 
</div></div></div>
```

in the cases where the above does not apply is not beneath me. :stuck_out_tongue_winking_eye:

Reply:
shudder